### PR TITLE
[SPARK-46154][INFRA] Add a new Daily testing GitHub Action job for Maven with Java 21

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims add a new Daily testing GitHub Action job for Maven with Java 21

### Why are the changes needed?
Need Daily testing for Maven with Java 21.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Monitor GA after merge

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -283,6 +283,14 @@ jobs:
         name: unit-tests-log-${{ matrix.modules }}-${{ matrix.comment }}-${{ matrix.java }}-${{ matrix.hadoop }}-${{ matrix.hive }}
         path: "**/target/unit-tests.log"
 
+  maven-test:
```
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:286]: will revert after test

**Comment by LuciferYang** [comment]
https://github.com/LuciferYang/spark/actions/runs/7028585225/job/19124783706

**Comment by LuciferYang** [comment]
image

The installed Java version is correct.

**Comment by LuciferYang** [comment]
cc @zhengruifeng

**Review by zhengruifeng — APPROVED** [review]

**Comment by zhengruifeng** [comment]
LGTM

**Comment by LuciferYang** [comment]
Thanks @zhengruifeng

**Comment by LuciferYang** [comment]
> https://github.com/LuciferYang/spark/actions/runs/7028585225/job/19124783706

image

all maven test passed, let me merge this and monitor GA tomorrow

**Comment by LuciferYang** [comment]
Merged into master. Thanks @zhengruifeng ~

**Pull request closed without merging**